scripts/emulate_app.sh

```diff
@@ -177,35 +178,43 @@ if [ ! -d "./android/app" ]; then
     echo "Try './scripts/$(basename $0)' from zingo-mobile root directory." >&2
     exit 1
 fi
-cd android
 
-echo -e "\nInstalling latest build tools, platform tools, and platform..."
-sdkmanager --install 'build-tools;33.0.2' platform-tools
+cd android
+# Kill all emulators
+../scripts/kill_emulators.sh
 
-echo "Installing latest emulator..."
-sdkmanager --install emulator --channel=0
+test_report_dir="app/build/outputs/emulate_app_reports/${abi}"
+rm -rf "${test_report_dir}"
+mkdir -p "${test_report_dir}"
 
-echo "Installing system image..."
 avd_name="${avd_skin}-android-${api_level}_${api_target}_${arch}"
 sdk="system-images;android-${api_level};${api_target};${arch}"
-sdkmanager --install "${sdk}"
-sdkmanager --licenses
-
-# Kill all emulators
-../scripts/kill_emulators.sh
+platform="platforms;android-${api_level}"
 
 if [[ $create_snapshot == true ]]; then
+    sdkmanager --licenses
+
+    echo -e "\nInstalling latest build tools, platform tools, and platform..."
+    sdkmanager --install 'build-tools;33.0.2' platform-tools emulator
+
+    echo "Installing latest emulator..."
+    sdkmanager --install emulator --channel=0
+
+    echo "Installing system image..."
+    sdkmanager --install "${sdk}"
+    sdkmanager --install "${platform}"
+    
     echo -e "\nCreating AVD..."
     echo no | avdmanager create avd --force --name "${avd_name}" --package "${sdk}" --device "${avd_skin}"
 
     echo -e "\n\nWaiting for emulator to launch..."
-    emulator -avd "${avd_name}" -netdelay none -netspeed full -no-window -no-audio -gpu swiftshader_indirect -no-boot-anim \
-        -no-snapshot-load -port 5554 &> /dev/null &
+    emulator -avd "${avd_name}" -netdelay none -netspeed full -no-window -no-audio -no-boot-anim -no-snapshot-load -port 5554 &> "${test_report_dir}/emulator.txt" &
```

Comment:
i don't like the create snapshot overriding test data or deleting previous test data. can we move the test directory stuff back to where it was and have a similar directory for creating snapshots if we want to capture the printouts?